0027-safely-recommend-actions.adoc

```diff
@@ -0,0 +1,173 @@
+= COMIT Protocol Actions
+Tobin C. Harding <tobin@coblox.tech>;
+:toc:
+:revdate: 2020-08-07
++ Created on: {revdate} +
+
+== Introduction
+
+In order for two parties to exchange assets (swap) using the COMIT protocol each party must take certain pre-defined actions at certain times. A swap is guaranteed to be atomic if, and only if, the correct sequence of actions is followed. It is therefore paramount that a user knows what stage the swap is up to and what step they should take next. Additionally, at any stage, each party should be able to answer the following question: "If I do the next step is there time, for myself and the counterparty, to do all subsequent steps?".
+
+== Swap State Machine
+
+A successful COMIT atomic swap (happy path) may be expressed as a series of state transitions:
+
+        Created --> Alpha Funded --> Beta Funded --> Beta Redeemed --> Alpha Redeemed --> Done
+
+Each transition is associated with a specific action named accordingly e.g., 'fund alpha' transitions from created to alpha funded.
+
+For each state transition there is a time requirement, henceforth called the transition period. The total time requirement for a swap is the sum of the transition periods. A transition period is effected by the state we are transitioning from and the state we are transitioning to, each transition period therefore is calculated differently.
+
+== Transition Period
+
+There exist a number of factors, that when combined, represent the transition period.
+
+.Considerations
+* Time to Act (`TTA`)
+* Time to Inclusion (`TTI`)
+* Time to Final (`TTF`)
+* HTLC Expiry time
+
+=== Time to Act
+
+For each actor there is a time frame within which they can be expected to act.
+
+Different users will have different response times e.g., a long running daemon is likely to respond to swap transitions more quickly than a user who is manually checking an application before doing each action. We define Time To Act (`TTA`) as the window of time within which the counterparty can expected to take the next action.
```

Comment:
>We define Time To Act (TTA) as the window of time within which the counterparty can expected to take the next action.

How do you justify the TTA?

Reply:
I made an educated guess. I specifically am not investing time into picking perfect values for this project, rather the focus is on making sure every value that would be needed to configure this is present.